torch/_dynamo/convert_frame.py

```diff
@@ -539,6 +543,7 @@ def compile_inner(
                     "Restarting analysis due to %s",
                     LazyString(format_traceback_short, e.__traceback__),
                 )
+                wasted_compile_time += time.time() - attempt_start_time
```

Comment:
It might be more robust to just compute only three timestamps: start, end, and start of the last restart before we succeeded.